umbral/curve.py

```diff
@@ -98,6 +99,13 @@ def field_order_size_in_bytes(self) -> int:
             self.__field_order_size_in_bytes = (size_in_bits + 7) // 8
         return self.__field_order_size_in_bytes
 
+    @property
+    def group_order_size_in_bytes(self) -> int:
+        if self.__group_order_size_in_bytes is None:
```

Comment:
Why are you checking this here? Python handles the caching when you do `@property`.

Reply:
It seems it doesn't. https://stackoverflow.com/questions/6428723/python-are-property-fields-being-cached-automatically